python/ray/data/tests/test_hudi.py

```diff
@@ -0,0 +1,110 @@
+import os
+import zipfile
+
+import pyarrow as pa
+import pytest
+from pytest_lazyfixture import lazy_fixture
+
+import ray
+from ray.data.datasource.path_util import (
+    _resolve_paths_and_filesystem,
+    _unwrap_protocol,
+)
+from ray.data.tests.conftest import *  # noqa
+from ray.data.tests.mock_http_server import *  # noqa
+from ray.tests.conftest import *  # noqa
+
+PYARROW_LE_11_0_0 = tuple(
+    int(s) for s in pa.__version__.split(".") if s.isnumeric()
+) < (
+    11,
+    0,
+    0,
+)
+pytestmark = pytest.mark.skipif(
+    PYARROW_LE_11_0_0, reason="hudi only supported if pyarrow >= 11.0.0"
+)
+
+
+def _extract_testing_table(fixture_path: str, table_dir: str, target_dir: str) -> str:
+    with zipfile.ZipFile(fixture_path, "r") as zip_ref:
+        zip_ref.extractall(target_dir)
+    return os.path.join(target_dir, table_dir)
+
+
+@pytest.mark.parametrize(
+    "fs,data_path",
+    [
+        (None, lazy_fixture("local_path")),
+        (lazy_fixture("local_fs"), lazy_fixture("local_path")),
+    ],
+)
+def test_read_hudi_simple_cow_table(ray_start_regular_shared, fs, data_path):
+    setup_data_path = _unwrap_protocol(data_path)
+    target_testing_dir = os.path.join(setup_data_path, "test_hudi")
+    fixture_path, _ = _resolve_paths_and_filesystem(
+        "example://hudi-tables/0.x_cow_partitioned.zip", fs
+    )
```

Comment:
Not really a fan of committing opaque binary blobs to the repo.

Can we instead create simple Hudi table manually, and then try to read it back using Ray Data?

Reply:
agree. this is intermediate solution as we don't support write capability in hudi-rs yet. Once we do, we can switch to generating testing tables